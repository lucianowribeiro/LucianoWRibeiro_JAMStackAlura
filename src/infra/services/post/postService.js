import isStagingEnv from '../../env/isStagingEnv';
import HttpClient from '../../http/HttpClient';
import authService from '../auth/authService';

const BASE_URL = isStagingEnv
  ? 'https://instalura-api-git-master-omariosouto.vercel.app'
  : 'https://instalura-api-omariosouto.vercel.app';

const postService = {
  async post(data, HttpClientModule = HttpClient, authServiceModule = authService) {
    try {
      const token = await authServiceModule().getToken();
      const response = await HttpClientModule(`${BASE_URL}/api/posts`, {
        method: 'POST',
        body: data,
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      return response;
    } catch (error) {
      return null;
    }
  },

};

export default postService;
