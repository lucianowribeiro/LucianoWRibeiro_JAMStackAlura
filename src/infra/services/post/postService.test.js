import postService from './postService';

const dataFake = {
  _id: 'some id',
  photoUrl: 'some url',
  description: 'some description',
  filter: 'some filter',
  user: 'some user',
  likes: [],
  createdAt: 'some data',
  updatedAt: 'some data 2',
  __v: 0,
};

async function HttpClientModule() {
  return {
    data: dataFake,
  };
}

describe('postService', () => {
  describe('post()', () => {
    describe('when user post an image', () => {
      describe('and succeed', () => {
        test('store the post data', async () => {
          const { data } = await postService.post({
            photoUrl: 'some url',
            description: 'some description',
            filter: 'some filter',
          }, HttpClientModule);
          await expect(data).toMatchSnapshot();
          await expect(data.filter).toBe(dataFake.filter);
        });
      });
      describe('and it fails', () => {
        test('return null', async () => {
          const response = await postService.post({});
          await expect(response).toMatchSnapshot();
          await expect(response).toBeNull();
        });
      });
    });
  });
});
