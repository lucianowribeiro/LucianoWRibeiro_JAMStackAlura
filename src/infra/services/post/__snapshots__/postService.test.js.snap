// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postService post() when user post an image and it fails return null 1`] = `null`;

exports[`postService post() when user post an image and succeed store the post data 1`] = `
Object {
  "__v": 0,
  "_id": "some id",
  "createdAt": "some data",
  "description": "some description",
  "filter": "some filter",
  "likes": Array [],
  "photoUrl": "some url",
  "updatedAt": "some data 2",
  "user": "some user",
}
`;
